zio-http/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -273,6 +274,202 @@ object HttpCodec
+    Sub3 <: Value: ClassTag,
+    Sub4 <: Value: ClassTag,
+    Sub5 <: Value: ClassTag,
+    Sub6 <: Value: ClassTag,
+    Sub7 <: Value: ClassTag,
+  ](
+    codec1: HttpCodec[AtomTypes, Sub1],
+    codec2: HttpCodec[AtomTypes, Sub2],
+    codec3: HttpCodec[AtomTypes, Sub3],
+    codec4: HttpCodec[AtomTypes, Sub4],
+    codec5: HttpCodec[AtomTypes, Sub5],
+    codec6: HttpCodec[AtomTypes, Sub6],
+    codec7: HttpCodec[AtomTypes, Sub7],
+  ): HttpCodec[AtomTypes, Value] =
+    (codec1 | codec2 | codec3 | codec4 | codec5 | codec6 | codec7).transformOrFail(
+      either =>
+        Right(either.left.map(_.left.map(_.left.map(_.left.map(_.left.map(_.merge).merge).merge).merge).merge).merge),
+      (value: Value) =>
+        value match {
+          case sub1: Sub1 => Right(Left(Left(Left(Left(Left(Left(sub1)))))))
+          case sub2: Sub2 => Right(Left(Left(Left(Left(Left(Right(sub2)))))))
+          case sub3: Sub3 => Right(Left(Left(Left(Left(Right(sub3))))))
+          case sub4: Sub4 => Right(Left(Left(Left(Right(sub4)))))
+          case sub5: Sub5 => Right(Left(Left(Right(sub5))))
+          case sub6: Sub6 => Right(Left(Right(sub6)))
+          case sub7: Sub7 => Right(Right(sub7))
+          case _          => Left(s"Unexpected error type")
+        },
+    )
+
+  def enumeration[
+    AtomTypes,
+    Value,
+    Sub1 <: Value: ClassTag,
+    Sub2 <: Value: ClassTag,
+    Sub3 <: Value: ClassTag,
+    Sub4 <: Value: ClassTag,
+    Sub5 <: Value: ClassTag,
+    Sub6 <: Value: ClassTag,
+    Sub7 <: Value: ClassTag,
+    Sub8 <: Value: ClassTag,
+  ](
+    codec1: HttpCodec[AtomTypes, Sub1],
+    codec2: HttpCodec[AtomTypes, Sub2],
+    codec3: HttpCodec[AtomTypes, Sub3],
+    codec4: HttpCodec[AtomTypes, Sub4],
+    codec5: HttpCodec[AtomTypes, Sub5],
+    codec6: HttpCodec[AtomTypes, Sub6],
+    codec7: HttpCodec[AtomTypes, Sub7],
+    codec8: HttpCodec[AtomTypes, Sub8],
+  ): HttpCodec[AtomTypes, Value] =
+    (codec1 | codec2 | codec3 | codec4 | codec5 | codec6 | codec7 | codec8).transformOrFail(
+      either =>
+        Right(
+          either.left
+            .map(_.left.map(_.left.map(_.left.map(_.left.map(_.left.map(_.merge).merge).merge).merge).merge).merge)
+            .merge,
+        ),
+      (value: Value) =>
+        value match {
+          case sub1: Sub1 => Right(Left(Left(Left(Left(Left(Left(Left(sub1))))))))
+          case sub2: Sub2 => Right(Left(Left(Left(Left(Left(Left(Right(sub2))))))))
+          case sub3: Sub3 => Right(Left(Left(Left(Left(Left(Right(sub3)))))))
+          case sub4: Sub4 => Right(Left(Left(Left(Left(Right(sub4))))))
+          case sub5: Sub5 => Right(Left(Left(Left(Right(sub5)))))
+          case sub6: Sub6 => Right(Left(Left(Right(sub6))))
+          case sub7: Sub7 => Right(Left(Right(sub7)))
+          case sub8: Sub8 => Right(Right(sub8))
+          case _          => Left(s"Unexpected error type")
+        },
+    )
+
+  def statusAndContent[Body](status: zio.http.Status)(implicit
```

Comment:
Not sure about this one, but perhaps:

```suggestion
  def error[Body](status: zio.http.Status)(implicit
```

Also we could think about putting it inside `StatusCodecs`, although I'm not 100% sure about that one either.

Reply:
It is not only a status codec, so I would leave it there